plugin/pkg/auth/authenticator/request/basicauth/basicauth.go

```diff
@@ -33,11 +36,52 @@ func New(auth authenticator.Password) *Authenticator {
 	return &Authenticator{auth}
 }
 
+type Credentials struct {
```

Comment:
these belong in a separate file, implemented as a authenticator.Password. See packages under https://github.com/kubernetes/kubernetes/tree/master/plugin/pkg/auth/authenticator/password